src/models/language_model.cc

```diff
@@ -298,30 +298,49 @@ namespace ctranslate2 {
 
     EncoderForwardOutput
     SequenceEncoderReplica::forward(const std::vector<std::vector<std::string>>& tokens) {
+      std::vector<std::vector<size_t>> token_type_ids;
+      return forward(tokens, token_type_ids);
+    }
+    
+    EncoderForwardOutput
+    SequenceEncoderReplica::forward(const std::vector<std::vector<std::string>>& tokens, const std::vector<std::vector<size_t>>& token_type_ids) {
       const auto& vocabulary = _model->get_vocabulary();
-      return forward(vocabulary.to_ids(tokens));
+      return forward(vocabulary.to_ids(tokens), token_type_ids);
     }
 
     EncoderForwardOutput
     SequenceEncoderReplica::forward(const std::vector<std::vector<size_t>>& ids) {
+      std::vector<std::vector<size_t>> token_type_ids;
+      return forward(ids, token_type_ids);
+    }
+    
+    EncoderForwardOutput
+    SequenceEncoderReplica::forward(const std::vector<std::vector<size_t>>& ids, const std::vector<std::vector<size_t>>& token_type_ids) {
       StorageView lengths;
       StorageView input_ids = layers::make_sequence_inputs(ids, Device::CPU, 1, &lengths);
-      return forward(input_ids, lengths);
+      return forward(input_ids, lengths, token_type_ids);
     }
 
     EncoderForwardOutput
     SequenceEncoderReplica::forward(const StorageView& ids, const StorageView& lengths) {
+      std::vector<std::vector<size_t>> token_type_ids;
+      return forward(ids, lengths, token_type_ids);
+    }
+    
+    EncoderForwardOutput
+    SequenceEncoderReplica::forward(const StorageView& ids, const StorageView& lengths, const std::vector<std::vector<size_t>>& token_type_ids) {
       PROFILE("SequenceEncoderReplica::forward");
       const auto& model = *this->model();
       const auto device = model.device();
       const auto scoped_device_setter = model.get_scoped_device_setter();
 
+      StorageView input_token_type_ids = layers::make_sequence_inputs(token_type_ids, Device::CPU, 1, nullptr);
```

Comment:
Directly create the storage on the target device.

```suggestion
      StorageView input_token_type_ids = layers::make_sequence_inputs(token_type_ids, device);
```